// react-highlight-js/lib/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _highlight = _interopRequireDefault(require("highlight.js"));

var _react = _interopRequireDefault(require("react"));

var _reactDom = _interopRequireDefault(require("react-dom"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

class Highlight extends _react.default.Component {
  componentDidMount() {
    this.highlightCode();
  }

  componentDidUpdate() {
    this.highlightCode();
  }

  highlightCode() {
    const domNode = _reactDom.default.findDOMNode(this);

    const nodes = domNode.querySelectorAll('pre code');
    let i;

    for (i = 0; i < nodes.length; i++) {
      _highlight.default.highlightBlock(nodes[i]);
    }
  }

  render() {
    const {
      children,
      className,
      element,
      innerHTML
    } = this.props;
    let Element = element ? _react.default.DOM[element] : null;

    if (innerHTML) {
      if (!Element) {
        Element = _react.default.DOM.div;
      }

      return Element({
        dangerouslySetInnerHTML: {
          __html: children
        },
        className: className || null
      }, null);
    } else {
      if (Element) {
        return Element({
          className
        }, children);
      } else {
        return _react.default.createElement("pre", null, _react.default.createElement("code", {
          className: className
        }, " ", children));
      }
    }
  }

}

Highlight.defaultProps = {
  innerHTML: false,
  className: null,
  element: null
};
var _default = Highlight;
exports.default = _default;
